packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -466,6 +469,10 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
               hasEnv = true
             }
             return
+          } else if (templateLiteralRE.test(rawUrl)) {
+            // Only static template literal will into this branch.
+            // It has variables will processed in importMetaGlob.ts
+            specifier = rawUrl.replace(templateLiteralRE, '$1')
```

Comment:
I don't think this is correct.
Dynamic import with `/* @vite-ignore */` or dynamic imports that cannot be handled with the dynamic-import plugin (e.g. ``import(`${startsWithVariable}/foo`)``) will be still there.
https://stackblitz.com/edit/vitejs-vite-fnln8a?file=main.js&terminal=dev
![image](https://github.com/vitejs/vite/assets/49056869/2d517b93-3033-4206-8b27-8392bb978196)

So `templateLiteralRE.test(rawUrl)` will match `` `foo${va}` ``
https://stackblitz.com/edit/node-ptwy2i?file=index.js

Reply:
Ah, no, I think I'm misunderstanding here...

Reply:
Oh, I realize that. A dynamic import that doesn't conform to the rules will still get in here. So do I just need to exclude paths containing variables here?

Reply:
> So do I just need to exclude paths containing variables here?

I think so 👍